src/avt/Expressions/CMFE/avtConnCMFEExpression.C

```diff
@@ -183,12 +188,81 @@ avtConnCMFEExpression::ExecuteTree(avtDataTree_p in1, avtDataTree_p in2,
         vtkDataSet *in_ds1 = in1->GetDataRepresentation().GetDataVTK();
         vtkDataSet *in_ds2 = in2->GetDataRepresentation().GetDataVTK();
 
+        vtkIdType nRealCells1 = in_ds1->GetNumberOfCells();
+        vtkIdType nRealCells2 = in_ds2->GetNumberOfCells();
+        vtkIdType nRealPoints1 = in_ds1->GetNumberOfPoints();
+        vtkIdType nRealPoints2 = in_ds2->GetNumberOfPoints();
+
+        //
+        // Check for the case where we are mapping between a point mesh
+        // and a polyhedral mesh and the number of polyhedra is the same
+        // as the number of points.
+        //
+        bool handlePointPolyhedralMapping = false;
+        if (nRealCells1 != nRealCells2 &&
+            in_ds1->GetDataObjectType() == VTK_UNSTRUCTURED_GRID &&
+            in_ds2->GetDataObjectType() == VTK_UNSTRUCTURED_GRID)
+        {
+            debug5 << "avtConnCMFEExpression::ExecuteTree: The cell counts"
+                   << " don't match and both grids are unstructured." << endl;
+            //
+            // Get the real number of cells in each dataset.
+            //
+            vtkDataArray *orig1 =
+                in_ds1->GetCellData()->GetArray("avtOriginalCellNumbers");
```

Comment:
So, without looking in detail, it seems like this is coded for the arb. poly<-->point case *only*. Does it handle the zoo-type cell<-->point case also? Seems like it should as the later is just a special case of the former where number of orig. cells to actual cells is 1:1.

Reply:
The case of mapping between zoo-type meshes and point meshes always worked when the number of cells and points matched. The new coding handles the case where we have 2 meshes where the number of cells doesn't match by using the original cells array.